providers/apis/polymarket/api_handler_test.go

```diff
@@ -0,0 +1,157 @@
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			url, err := h.CreateURL(tc.pts)
+			if tc.expErr != "" {
+				require.ErrorContains(t, err, tc.expErr)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, url, tc.expectedURL)
+			}
+		})
+	}
+}
+
+func TestParseResponse(t *testing.T) {
+	id := candidateWinsElectionToken
+	handler, err := NewAPIHandler(DefaultAPIConfig)
+	require.NoError(t, err)
+	testCases := []struct {
+		name             string
+		noError          bool
+		ids              []types.ProviderTicker
+		responseBody     string
+		expectedResponse types.PriceResponse
+	}{
+		{
+			name:         "happy case",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken},
+			noError:      true,
+			responseBody: `{ "price": "0.45" }`,
+			expectedResponse: types.NewPriceResponse(
+				types.ResolvedPrices{
+					id: types.NewPriceResult(big.NewFloat(0.45), time.Now().UTC()),
+				},
+				nil,
+			),
+		},
+		{
+			name:         "1.00 should resolve to 0.999...",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken},
+			noError:      true,
+			responseBody: `{ "price": "1.00" }`,
+			expectedResponse: types.NewPriceResponse(
+				types.ResolvedPrices{
+					id: types.NewPriceResult(big.NewFloat(priceAdjustment), time.Now().UTC()),
+				},
+				nil,
+			),
+		},
+		{
+			name:         "too many IDs",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken, candidateWinsElectionToken},
+			responseBody: ``,
+			expectedResponse: types.NewPriceResponseWithErr(
+				[]types.ProviderTicker{candidateWinsElectionToken, candidateWinsElectionToken},
+				providertypes.NewErrorWithCode(
+					fmt.Errorf("expected 1 ticker, got 2"),
+					providertypes.ErrorInvalidResponse,
+				),
+			),
+		},
+		{
+			name:         "invalid JSON",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken},
+			responseBody: `{"price": "0fa3adk"}"`,
+			expectedResponse: types.NewPriceResponseWithErr(
+				[]types.ProviderTicker{candidateWinsElectionToken},
+				providertypes.NewErrorWithCode(fmt.Errorf("failed to convert %q to float", "0fa3adk"), providertypes.ErrorFailedToDecode),
+			),
+		},
+		{
+			name:         "bad price - max",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken},
+			responseBody: `{"price": "1.0001"}"`,
+			expectedResponse: types.NewPriceResponseWithErr(
+				[]types.ProviderTicker{candidateWinsElectionToken},
+				providertypes.NewErrorWithCode(fmt.Errorf("price exceeded 1.00"), providertypes.ErrorInvalidResponse),
+			),
+		},
+		{
+			name:         "bad price - negative",
+			ids:          []types.ProviderTicker{candidateWinsElectionToken},
+			responseBody: `{"price": "-0.12"}"`,
+			expectedResponse: types.NewPriceResponseWithErr(
+				[]types.ProviderTicker{candidateWinsElectionToken},
+				providertypes.NewErrorWithCode(fmt.Errorf("price must be greater than 0.00"), providertypes.ErrorInvalidResponse),
+			),
```

Comment:
Add test case with price 0

Reply:
added here [add adjustment to 0.00 price](https://github.com/skip-mev/slinky/pull/629/commits/fc2a4769b7df77fe3b91571fe5859c91cbf5ac64)